homeassistant/components/aidot/config_flow.py

```diff
@@ -0,0 +1,181 @@
+"""Config flow for Aidot integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aidot.login_const import DEFAULT_COUNTRY_NAME, SUPPORTED_COUNTRY_NAMES
+from aidot.login_control import LoginControl
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_COUNTRY, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_CHOOSE_HOUSE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: HomeAssistant, data: dict) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    if len(data["host"]) < 3:
+        raise InvalidHost
+    return {"title": data["host"]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle aidot config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self.__login_control = LoginControl()
+        self.login_response: dict[Any, Any] = {}
+        self.accessToken = ""
+        self.house_list: list[Any] = []
+        self.device_list: list[Any] = []
+        self.product_list: list[Any] = []
+        self.selected_house: dict[Any, Any] = {}
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                # get ContryCode
+                selected_contry_name = user_input[CONF_COUNTRY]
+                self.__login_control.change_country_name(selected_contry_name)
+
+                self.login_response = await self.__login_control.async_post_login(
+                    self.hass,
+                    user_input[CONF_USERNAME],
+                    user_input[CONF_PASSWORD],
+                )
+                if self.login_response is None:
+                    errors["base"] = "login_failed"
+                    return self.async_show_form(
+                        step_id="user", data_schema=None, errors=errors
+                    )
+                self.accessToken = self.login_response["accessToken"]
```

Comment:
```suggestion
                self.accessToken = self.login_response[CONF_ACCESS_TOKEN]
```
also everywhere else, where you're storing things in the entry